casadm/cas_lib.c

```diff
@@ -1825,38 +1824,39 @@ bool _device_mounts_detected(int cache_id, int core_id)
 	}
 
 	while ((mstruct = getmntent(mtab)) != NULL) {
-		error = strcmp_s(mstruct->mnt_fsname, cmplen, dev_buf, &no_match);
+		error = strcmp_s(mstruct->mnt_fsname, cmplen, pattern, &no_match);
 		/* mstruct->mnt_fsname is /dev/... block device path, not a mountpoint */
 		if (error != EOK)
 			return false;
 		if (no_match)
 			continue;
 
-		if (core_id < 0) {
-			cas_printf(LOG_ERR,
-				   "Can't stop cache instance %d. Device %s is mounted!\n",
-				   cache_id, mstruct->mnt_fsname);
-		} else {
-			cas_printf(LOG_ERR,
-				   "Can't remove core %d from cache %d."
-				   " Device %s is mounted!\n",
-				   core_id, cache_id, mstruct->mnt_fsname);
-		}
-
 		return true;
 	}
 
 	return false;
 }
 
-bool cache_mounts_detected(int cache_id)
+void print_mounted_devices(const char *pattern, int cmplen)
 {
-	return _device_mounts_detected(cache_id, -1);
-}
+	FILE *mtab;
+	struct mntent *mstruct;
+	int no_match = 0, error = 0;
 
-bool core_mounts_detected(int cache_id, int core_id)
-{
-	return _device_mounts_detected(cache_id, core_id);
+	mtab = setmntent("/etc/mtab", "r");
+	if (!mtab) {
+		/* should exist, but if /etc/mtab not found we cannot print mounted devices */
+		return;
+	}
+
+	while ((mstruct = getmntent(mtab)) != NULL) {
+		error = strcmp_s(mstruct->mnt_fsname, cmplen, pattern, &no_match);
+		/* mstruct->mnt_fsname is /dev/... block device path, not a mountpoint */
+		if (error != EOK || no_match)
```

Comment:
Again, the error here gets ignored.

Reply:
If we get an error, then we cannot determine if the device is mounted or not -> let's at least try to remove it.